### PR TITLE
TST: make venv test fixture work on Windows and with PyPy

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

**Comment by FFY00** [comment]
Eh, I think it may be just be easier to use the `vitualenv` API.

**Comment by dnicolodi** [comment]
> I think it may be just be easier to use the `virtualenv` API.

This now works fine on all platforms with CPython. It still fails on PyPy but I'm not sure `virtualenv` helps there. Everything is failing now because I'm trying to extend the test for the example project to other Python versions and platforms.

**Comment by dnicolodi** [comment]
By the way, do we link to the example from anywhere in the documentation?

**Comment by dnicolodi** [comment]
This turned into a more general cleanup of the tests, but it is now ready for review.

I reviewed all XFAIL and SKIP tests. Now, other than test skipped for genuinely unimplemented features on a given platform, some tests are skipped only:

- on Python 3.7 because the extension filenames do not allow to derive the ABI tag (fixed in #191)
- on Linux and macOS when running with PyPy because creating a venv fails.

I'm still investigating the venv issue.

**Comment by dnicolodi** [comment]
Rebased. Let's see if this fixes the conda tests.

**Comment by dnicolodi** [comment]
The venv issues are caused by nox. I removed the use of nox from the CI setup. I think not using it actually makes things easier. Please have a look.

File: .github/workflows/tests.yml
Comment (FFY00) [inline .github/workflows/tests.yml:52]: This isn't needed in CI.

**Comment by FFY00** [comment]
> The venv issues are caused by nox. I removed the use of nox from the CI setup. I think not using it actually makes things easier. Please have a look.

This is not optimal, but good enough for now.

**Review by FFY00 — APPROVED** [review]

File: .github/workflows/tests.yml
Comment (FFY00) [reply .github/workflows/tests.yml:52]:
```suggestion
```

**Comment by dnicolodi** [comment]
@FFY00 Can I merge (well, rebase and merge) my own PRs once they are approved? The readthedocs check is a requirement for merging but something is still off with it, thus I cannot merge this.

**Comment by FFY00** [comment]
Yep. I will merge this one since the required checks are broken, I need to sort out the readthedocs situation.

**Pull request merged**
